[Do not merge] Add 2021-04-01 SRP api version

HimanshuChhabra opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

HimanshuChhabra committed:
Create 2021-04-01 SRP api version, no features

HimanshuChhabra committed:
Create 2021-04-01 SRP api version, no features

HimanshuChhabra committed:
Adding Blob Inventory GA Swagger changes and examples

HimanshuChhabra committed:
Adding Blob Inventory GA Swagger changes and examples

HimanshuChhabra committed:
Add Container VLW Swagger changes

HimanshuChhabra committed:
Add Container VLW Swagger changes

HimanshuChhabra committed:
Add Lease Share (Commit 1204774b: Merged PR 4370500: Swagger: deleted share with leased snapshot; lease share; share ACLs; list both deleted shares and snapshots; allow cross tenant replication; default share permission)

HimanshuChhabra committed:
Add Lease Share (Commit 1204774b: Merged PR 4370500: Swagger: deleted share with leased snapshot; lease share; share ACLs; list both deleted shares and snapshots; allow cross tenant replication; default share permission)

HimanshuChhabra committed:
Add custom-words, Add default responses for new apis

HimanshuChhabra committed:
Add custom-words, Add default responses for new apis

HimanshuChhabra committed:
Capitalize VNet rule state

HimanshuChhabra committed:
Capitalize VNet rule state

litchiyangMSFT committed:
hotfix on extended location hierarchy (#13943)

vicancy committed:
[Hub Generated] Public private branch 'wps' (#14002)

* Add webpubsub swagger

Add custom words

Update webpubsub.json

Update webpubsub.json

Resolve comments

Update webpubsub.json

resolve comments

Update operationId for existence check

[wps] update python generation section and client name for all languages (#1)

* add python stuff

* make title across languages the same

* Update version

* Update some constraints

* Update webpubsub.json

* rename to GetServiceStatus

* Update readme.md

* Update readme.md

* Update readme.md

mayank88mahajan committed:
[Hub Generated] Review request for Microsoft.SecurityInsights to add version preview/2021-03-01-preview (#13563)

* Copied from 2019-01-01-preview version

* Updated references to common types

* Add itemsSearchKey field to Watchlists

* Add missing label definition

* Make itemsSearchKey as required field

* Formatting fixes

* Fix Prettier and Avocado errors

* Fix Lint error (subscription level resource)

* Update description and example for CreateOrUpdate watchlists and bulk create watchlist items

* Update Api version in examples

chandrikagonuguntla committed:
[Hub Generated] Review request for Microsoft.RecoveryServices to add version stable/2021-03-01 (#13888)

* Adds base for updating Microsoft.RecoveryServices from version stable/2021-01-01 to version 2021-03-01

* Updates readme

* Updates API version in new specs and examples

* updated default package in readme.md to 2021-03

Co-authored-by: Chandrika Gonuguntla <chgonugu@microsoft.com>

ritwik8119 committed:
Updated required and default fields in Action Groups spec (#13677)

* Updated error response for Action Groups API

* spacing

* added missing fields to Action Group Resource spec

* Added missing fields for action group resource in older version

* Updated "resource" name

* Removed changes to errorResponse

* Removed errorObject

* format

* Updated required and default fields in Action groups spec

* Update default value to boolean

Co-authored-by: kai ru <69238381+kairu-ms@users.noreply.github.com>

* changes default value type to boolean

* undo webhook receiver changes

Co-authored-by: Ritwik Rajendra <rirajend@microsoft.com>
Co-authored-by: kai ru <69238381+kairu-ms@users.noreply.github.com>

wzhang373 committed:
Remove AML Compute File Server (#13944)

* Remove AML Compute File Server

* update

* update

* update

* update

maddieminn committed:
Cost Management: Reservation Recommendations API: Add multiple missing properties in specs (#13967)

* added totalCost, properties.resourceType, properties.subscriptionId, and eTag to various RiRec dfinitions

* moving eTag  property to right place

* remove duplicate description

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/storage.json:
```diff
@@ -4313,12 +4313,58 @@
+          "description": "This is a required field. This field specifies the fields and properties of the object to be included in the inventory. Valid values for this field for the blob object type include Name, Creation-Time, Last-Modified, Content-Length, Content-MD5, BlobType, AccessTier, AccessTierChangeTime, Expiry-Time, hdi_isfolder, Owner, Group, Permissions, Acl, Snapshot, VersionId, IsC ...
         }
       },
       "required": [
```
If Filter is not required, what's the default value for filter? like what blob type, blobprefix will be used?

HimanshuChhabra replied:
Updated the description it will make it clear

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/storage.json:
```diff
@@ -4313,12 +4313,58 @@
+          "items": {
+            "type": "string"
+          },
+          "description": "This is a required field. This field specifies the fields and properties of the object to be included in the inventory. Valid values for this field for the blob object type include Name, Creation-Time, Last-Modified, Content-Length, Content-MD5, BlobType, AccessTier, AccessTierChangeTime, Expiry-Time, hdi_isfolder, Owner, Group, Permissions, Acl, Snapshot, VersionId, IsC ...
```
In the description, Could you give more detail in description like following. Since SRP Doc is generated from swagger. We should include the info in SRP doc.
i. Which field is mandatory.
ii. Which field is required when specify "includeBlobVersions" or "includeSnapshots".
iii. Which field is only available with HNS enabled.
b. It should be better if we can revise the swagger to guild user to input the schemafields more easily, and avoid mistake. (anyway, I have not get a way to do so, @Juliehzl any idea?)

HimanshuChhabra replied:
I have updated the description, addressing all your points

blueww replied:
Miss:  'Name' is mandatory field.

HimanshuChhabra replied:
done

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/examples/StorageAccountGetBlobInventoryPolicy.json:
```diff
@@ -4,7 +4,7 @@
     "resourceGroupName": "res7687",
     "accountName": "sto9699",
     "blobInventoryPolicyName": "default",
-    "api-version": "2020-08-01-preview",
```
Just to confirm, the blob inventory policy don't have "Version" any more.
In the original GA design, it should have version as "1.0".

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/examples/StorageAccountGetBlobInventoryPolicy.json:
```diff
@@ -17,11 +17,11 @@
           "policy": {
```
Just to confirm, the blob inventory policy don't have "Version" any more.
In the original GA design, it should have version as "1.0".

HimanshuChhabra replied:
Version is not a customer-facing property

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/common.json:
```diff
@@ -0,0 +1,137 @@
+  "info": {
+    "title": "Storage",
+    "description": "Storage Resource Provider API Common Types",
+    "version": "2021-04-01"
```
The API version 2021-04-01 still not available in any region like canary.
Per @HimanshuChhabra "2021-04-01 ARM manifest rollout is currently in progress and will finish by thursday".
We should not merge this PR before the new API version is available on server.

zhangyan133 committed:
remove track1 configure (#14044)

Co-authored-by: Yan Zhang (WICRESOFT NORTH AMERICA LTD) <v-yanzhang@microsoft.com>

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/storage.json:
```diff
@@ -4313,12 +4313,58 @@
         "filters": {
           "$ref": "#/definitions/BlobInventoryPolicyFilter",
           "description": "An object that defines the filter set."
```
can we specify filter for container rule?
will the prefix match also work for container? (If so, please update prefix description)

HimanshuChhabra replied:
Filter cannot be specified for container, it can only be specified for blobs

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/storage.json:
```diff
@@ -4313,12 +4313,58 @@
+          "items": {
+            "type": "string"
+          },
+          "description": "This is a required field. This field specifies the fields and properties of the object to be included in the inventory. Valid values for this field for the blob object type include Name, Creation-Time, Last-Modified, Content-Length, Content-MD5, BlobType, AccessTier, AccessTierChangeTime, Expiry-Time, hdi_isfolder, Owner, Group, Permissions, Acl, Snapshot, VersionId, IsC ...
```
"Snapshot" must and only can specify when includeSnapshots = true?
"VersionId","IsCurrentVersion" must and only can specify when  includeBlobVersions= true?

If so, I will not open "Snapshot","VersionId","IsCurrentVersion" in schemafields to customer, but add it automaticlly when user specify "includeSnapshots" or "includeBlobVersions" as true

HimanshuChhabra replied:
yes this is correct, 
Snapshot is required when includeSnapshots = true and cannot be used if includeSnapshots = false
"VersionId","IsCurrentVersion  is required when includeBlobVersions= true and cannot be used if includeBlobVersions= false

nilgupta6 committed:
New API version (2021-02-10) for recovery services site recovery (#13525)

* Adding old API version(2018-07-10) to new folder for 2021-20-10

* all changes for swagger (2021-02-10)

* read me and custom word changes

* Ayesha's comments addressed

* adress PR comments from ayesha

* Adding suppressions

* fixing custom words

nilgupta6 committed:
[Hub Generated] Review request for Microsoft.RecoveryServices to add version stable/2021-02-10 (#13636)

* Adds base for updating Microsoft.RecoveryServices from version stable/2021-02-01 to version 2021-02-10

* Updates readme

* Updates API version in new specs and examples

* Update readme.md

changing read me to include date too in tag

* cahnging examples API version to 2021-02-10

* fix read me syntax

xyxyxyxyxyxy committed:
TaskStepProperties swagger fix (#13971)

kaiwenfeng4086 committed:
Adding "assignmentState" property to scope model in access review APIs (#14039)

RAY-316 committed:
iothub readme.python package version change (#14043)

* iothub readme.python package version change

* multiapiscript add

* clear output folder

* Update readme.python.md

* Update readme.python.md

* Update readme.python.md

Co-authored-by: msyyc <70930885+msyyc@users.noreply.github.com>

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/storage.json:
```diff
@@ -4313,12 +4313,58 @@
         }
       },
       "required": [
-        "filters"
```
As filter is not required now, will blobtype still be required for filter?
It seems not much valuable to make blobtype required in filter, as we can inventory blobs with all types by default.

HimanshuChhabra replied:
blobType is required when objectType is Blob. I have updated the Descrtiption

dependabot[bot] committed:
Bump y18n from 4.0.0 to 4.0.1 (#13704)

Bumps [y18n](https://github.com/yargs/y18n) from 4.0.0 to 4.0.1.
- [Release notes](https://github.com/yargs/y18n/releases)
- [Changelog](https://github.com/yargs/y18n/blob/master/CHANGELOG.md)
- [Commits](https://github.com/yargs/y18n/commits)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

HimanshuChhabra committed:
Update Descriptions for the Blob Inventory Properties.

HimanshuChhabra committed:
Update Descriptions for the Blob Inventory Properties.

yupwei68 committed:
update (#14061)

detienne20 committed:
package-templatespecs-2021-03-preview tag added (python) (#13958)

* Added tag

* nit: ordering

Co-authored-by: Dania Etienne <daetienn@microsoft.com>

ericshape committed:
Use 2014-04-01 replication link specs in V5 SDK (#14058)

* Use 2014-04-01 replication link specs in V5 SDK

* remove legacy file.

heaths committed:
Update cSpell.json for Key Vault 7.2 (storage) (#14060)

RAY-316 committed:
automanage readme t1 del (#14029)

* automanage readme t1 del

* automanage readme t1 del

* readme.md t1 config del

leniatgh committed:
add missing reference (#13904)

zhangyan133 committed:
databoxedge readme modification (#14063)

Co-authored-by: Yan Zhang (WICRESOFT NORTH AMERICA LTD) <v-yanzhang@microsoft.com>

zhangyan133 committed:
managedservices t2 onfig (#14064)

Co-authored-by: Yan Zhang (WICRESOFT NORTH AMERICA LTD) <v-yanzhang@microsoft.com>

nilgupta6 committed:
[Hub Generated] Review request for Microsoft.RecoveryServices to add version stable/2021-02-10 (#13635)

* Adds base for updating Microsoft.RecoveryServices from version stable/2021-01-01 to version 2021-02-10

* Updates readme

* Updates API version in new specs and examples

* fixing merge issue

* removing vault usages from 2021-02

* Adding back vault usages

* updating readme

maddieminn committed:
updated etag to eTag (#13968)

joshdmcc committed:
[Hub Generated] Review request for Microsoft.InstanceMetadataService to add version stable/2020-12-01 (#13931)

* Adds base for updating Microsoft.InstanceMetadataService from version stable/2020-10-01 to version 2020-12-01

* Updates readme

* Updates API version in new specs and examples

* Adding new spot VM fields

Co-authored-by: Joshua McCarthy <jmccarthy@microsoft.com>

msyache committed:
Add FaceAPI PersonDirectory spec updates to preview v1.0 (#13766)

* Add FaceAPI v1.0-preview

Add a preview version of FaceAPI specs intended for public preview.
Contains existing spec in v1.0 plus PersonDirectory feature changes.

* Fixes after running validation

* Fixing validation

* Fixes 2

* Fixes 3

* Add PersonDirectory examples and fix routes

* Fixes 4

* Fixes 5

* Fixes 5.5

Co-authored-by: Yang Chen <yachen4@microsoft.com>

yl-ms committed:
Update the name and format of a parameter for randomnumbers (#14017)

* Update the name and format of a parameter for randomnumbers

* Change the parameter name from 'bytes_length' back to 'bytesLength'

aditimalladi committed:
Amalladi/extendedlocation update (#13849)

* Updating Specification to remove 400 Badrequest from Swagger File

* Removing 400 status code from examples

* Removing private preview API version from Repo

* Removing 2020-07-15-privatepreview tag from ReadME files

Co-authored-by: Aditi <amalladi@microsoft.com>

zhangyan133 committed:
maps t2 config (#14075)

Co-authored-by: Yan Zhang (WICRESOFT NORTH AMERICA LTD) <v-yanzhang@microsoft.com>

zhangyan133 committed:
Revert "remove track1 configure (#14044)" (#14047)

This reverts commit 867165cc24d81c487ad186c998c46ba55f81c77b.

nilgupta6 committed:
adding role assignment (#14077)

ericshape committed:
add connectionPolicy spec back to V5 SDK (#14088)

zhangyan133 committed:
modification (#14087)

Co-authored-by: Yan Zhang (WICRESOFT NORTH AMERICA LTD) <v-yanzhang@microsoft.com>

amisi01 committed:
[Hub Generated] Review request for Microsoft.DocumentDB to add version preview/2021-04-01-preview (#13710)

* Adds base for updating Microsoft.DocumentDB from version preview/2021-03-01-preview to version 2021-04-01-preview

* Updates readme

* Updates API version in new specs and examples

* Adding new API for fetching backup information(such as latest restorable timestmap) for a resource. Also, adding startTime,endTime to restorable.json

* Adding new API for fetching backup information(such as latest restorable timestmap) for a resource. Also, adding startTime,endTime to restorable.json

* Adding new API for fetching backup information(such as latest restorable timestmap) for a resource. Also, adding startTime,endTime to restorable.json

* adding Services APIs

* more changes

* updated examples

* prettier and readme updates

* importing changes missed from old versions

* nits

* fixing semantic validation

* fixing linter

* fix

* more fixes

* adding list operation and default error response

* specifying format integer

* attempt to fix sdks

* address arm comments and fix ci

* fixes

* fix

* try to fix ci

* fix examples

* .\services.json

* adding additionalProperties back to ServiceResouce

* adding additionalProperties back to ServiceResouce

* fix

* fix sdks

* fix linter

Co-authored-by: Meha Kaushik <kaushik.meha@gmail.com>
Co-authored-by: Meha Kaushik <mekaushi@microsoft.com>

zhangyan133 committed:
media t2 config(wave4) (#14074)

* media t2 config

* add packageversion

Co-authored-by: Yan Zhang (WICRESOFT NORTH AMERICA LTD) <v-yanzhang@microsoft.com>

Commit:
[Hub Generated] Review request for Microsoft.Maps to add version stable/2021-02-01 (#13733)

* Adds base for updating Microsoft.Maps from version preview/2020-02-01-preview to version 2020-02-01

* Updates for 2021-02-01 api version swagger

* creator swagger fixes

* fix model examples

* nextLink, uniqueId

* build fixes

* fixing listkeys

* fix long running operations

* fixing spelling

* fixing async behavior and examples

* updating arm region

* reset package-lock

giakas committed:
[Hub Generated] Review request for Microsoft.Media to add version preview/2021-05-01-preview (#13838)

* New Readme Config File

* New Go Language Readme Config File

* New Azure AZ Readme Config File

* New Azure CLI Readme Config File

* New Typescript Language Readme Config File

* New Python Language Readme Config File

* New C# Language Readme Config File

* New AzureResourceSchema Readme Config File

* New Swagger Spec File

* New Swagger Example Spec File

* Copy Signed off 2021-03-01 swaggers here

* Update swaggers to 2021=05-01-preview

* add missing examples

* update title

* fix Operation id

* fix video analyzer examples

* update Video swagger

* add format for $top

* remove unnecessary example file

* use chackname definitions from types.json

* fix segment length readonly

* change getStreamingToken to listStreamingToken, getProvisioningToken to listProvisioningToken

* fix the example file name

* fix a readonly issue with a video property

* make type readonly

* edits to descriptions

* adding missing descriptions

* address review feedback,

* few cleanup for update class

Co-authored-by: Christopher Bennage <bennage@microsoft.com>

dramuy committed:
Include the preview version of resource changes and resources history… (#14084)

* Include the preview version of resource changes and resources history APIs

* Address the review feedback on releasing the new package version.

idear1203 committed:
Update readme for Synapse management SDK May release (#14054)

* Update readme for Synapse management SDK May release

* Add package-2021-05-preview

* Address comment

Co-authored-by: Dongwei Wang <dongwwa@microsoft.com>

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/file.json:
```diff
@@ -471,6 +452,13 @@
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "Optional. Valid values are: snapshots, leased-snapshots, none. The default value is snapshots. For 'snapshots', the file share is deleted including all of its file share snapshots. If the file share contains leased-snapshots, the deletion fails. For 'leased-snapshots', the file share is deleted included all of its file share snapshots (leased/unleased). For 'none', the ...
```
Please confirm the default value is snapshots on server.

zfchen95 replied:
Confirmed

vipinhas committed:
removing read only property, no longer in use - NOT a breaking change (#13812)

RAY-316 committed:
resources readme.python t2 config (#14091)

RAY-316 committed:
peering readme t2 config (#14106)

jorgecotillo committed:
Remove replicationLinks_legacy.json (#14095)

PR #14058 removed this file: replicationLinks_legacy.json, removing from the schema config to prevent schema generation failures.

yuzhangyi committed:
Support sheet index in excel (#14062)

* Enable excel in public swagger

* update

* rollback dataflow change

* Support sheet index in excel

Co-authored-by: zhanyu2014 <zhanyu@microsoft.com>

stuartko committed:
Adding 2021-05-01 api-version to Template Specs (no changes over last preview spec) (#14065)

* Copy existing 2021-03-01-preview specs to new 2021-05-01 stable version (no schema changes)

* Changing api-version within specs to 2021-05-01

* Adding package tag for new stable version to readme.md, but not making it a default (yet)

ityankel committed:
Fix SecurityInsights Go SDK versioning (#14030)

* Fix SecurityInsights Go SDK versioning

* rename stable api version tag

* remove 'only' from tags with full functionality

nschonni committed:
chore: Move typo supression in RecoveryServices (#14120)

tjegbejimba committed:
Add missing Zone mapping property to Microsoft.Resources (#13936)

* remove name validation from client side

* remove name check from deploymentScripts swagger

* remove name check from previous swagger versions

* remove pattern from resourcegroupname parameter

* add zoneMappings to ResoursesType

* backfill older versions of swagger with zonemapping property

kairu-ms committed:
[Application Insights] Resolve duplicated definition names to fix track2 SDK generation error (#13951)

* Fix duplicate name

* Update readme.md

* Update readme.md

* Update and rename workbookOperations_API.json to aiOperations_API.json

* Update readme.md

* Update workbooks_API.json

* Update readme.md

* Update readme.python.md

* Update readme.md

* Update readme.md

* Update readme.md

* Rename aiOperations_API.json to workbookOperations_API.json

* Update readme.md

* Update readme.md

* Update readme.md

* Update readme.md

* Update readme.md

* Update readme.md

* Update specification/applicationinsights/resource-manager/readme.python.md

* Update specification/applicationinsights/resource-manager/readme.python.md

* Update specification/applicationinsights/resource-manager/readme.python.md

hongxian committed:
Update swagger for SSIS-IR to support user-assigned managed identity (#14114)

Co-authored-by: Hongxian Yao <hoyao@microsoft.com>

RAY-316 committed:
postgresql readme t2 config (#14067)

giakas committed:
Dev mediaservices microsoft.media 2021 05 01 (#13676)

* Adds base for updating Microsoft.Media from version stable/2020-05-01 to version 2021-05-01

* Updates readme

* Updates API version in new specs and examples

* [AMS] new 2021-05-01 release for Account service in AMS

* add missing example files

* fix the api version and missing example files

* Create a new Update definition to fix the "R2016 - PatchBodyParametersSchema "

* fix respnse

* fix update example

* add the 2016 supression for new tag as well.

HimanshuChhabra committed:
Address Description change to make BI schema fields value Name as always required.

HimanshuChhabra committed:
Address Description change to make BI schema fields value Name as always required.

HimanshuChhabra committed:
Update BI descrption

HimanshuChhabra committed:
Update BI descrption

Juliehzl commented:
@HimanshuChhabra Could you pay attention to breaking changes detected cross api version and check it is by design or by mistake?https://github.com/Azure/azure-rest-api-specs/pull/14038#issuecomment-823539088

RAY-316 committed:
mixedreality t2 readme config(wave4) (#14105)

* mixedreality t2 readme config

* t2 config del

msyyc committed:
[python] t2-confidentialledger config (#14144)

* Update readme.python.md

* Update readme.python.md

msyyc committed:
Update readme.python.md (#14145)

vakuncha committed:
update elastic swagger  (#14076)

* Update elastic.json

* added new examples

* taking the files from rpsaas master

* added suppression in readme

nichatur committed:
[Cosmos DB] Introduce new api version 2021-04-15 (#13873)

* [Cosmos DB] Introduce new api version 2021-04-15

Initial Commit

* Updated for 2021-04-15

* Dummy Commit

* Fixing privatelinkresources.json

* Removing ErrorResponseUpdatedFormat

* Suppressing systemData check

Commit:
Relay - Fix typo in the NetworkRuleSet path (#13962)

* Relay - Fix typo in the NetworkRuleSet path

* Suppression - DefaultErrorResponseSchema - we will be Implementing in new API version

msHich committed:
[Hub Generated] Review request for Microsoft.Portal to add version preview/2015-08-01-preview (#12732)

* Fixing properties in dashboard parts

* Updating Dashboard part metadata

* Updating based on prettier fixes

Co-authored-by: Himanshu Chandel <hich@microsoft.com>

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/file.json:
```diff
@@ -538,6 +526,62 @@
+          }
+        ],
+        "responses": {
+          "200": {
```
Acquire lease return 204 (created), instead if 200.
Please make sure server and swagger behavior are aligned.

zfchen95 replied:
I don't see 204 from API logs. I think it is 201 for acquire lease and 202 for break the lease. Will update the response code here.

blueww replied:
Yes, it should be 201. Please add it. and make sure all possible return codes are covered.

RAY-316 committed:
resources t1 config add (#14146)

* t1 config add

* package version

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/file.json:
```diff
@@ -798,6 +842,55 @@
           "readOnly": true,
           "description": "The approximate size of the data stored on the share. Note that this value may not include all recently created or recently resized files."
         },
+        "leaseStatus": {
```
After Acquire lease on a share, then get the share, the LeaseDuration, LeaseState, LeaseStatus seems won't return.

zfchen95 replied:
Fixed.

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/storage.json:
```diff
@@ -0,0 +1,4549 @@
+          "items": {
+            "type": "string"
+          },
+          "description": "An array of predefined enum values. Valid values include blockBlob, appendBlob, pageBlob. Hns accounts does not support pageBlobs. This field is required when definition.objectType property is set to 'Blob'."
```
Per the description "This field is required when definition.objectType property is set to 'Blob'"
But we can add blob inventory policy with ObjectType as Blob, but no objectType property , like following.
So please make sure if the description incorrect, or server behavior incorrect?

```
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/{subscription-id}/resourceGroups/***/providers/Microsoft.Storage/storageAccounts/***/inventoryPolicies/default?api-version=2021-04-01

Headers:
x-ms-client-request-id        : a842fecd-143b-488f-bfb1-f7ba83dd3ab7
accept-language               : en-US

Body:
{
  "properties": {
    "policy": {
      "enabled": false,
      "rules": [
        {
          "enabled": true,
          "name": "test2",
          "destination": "weitest1",
          "definition": {
            "format": "Parquet",
            "schedule": "Weekly",
            "objectType": "Blob",
            "schemaFields": [
              "Name",
              "Last-Modified",
              "Metadata",
              "LastAccessedTime"
            ]
          }
        }
      ],
      "type": "Inventory"
    }
  }
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
Pragma                        : no-cache
x-ms-request-id               : 8bda505f-de10-4cd8-a0f2-0bb223160659
Strict-Transport-Security     : max-age=31536000; includeSubDomains
x-ms-ratelimit-remaining-subscription-writes: 1199
x-ms-correlation-request-id   : fe0d7c84-d427-40e3-b6d2-1c79e2e09995
x-ms-routing-request-id       : SOUTHEASTASIA:20210427T065956Z:fe0d7c84-d427-40e3-b6d2-1c79e2e09995
X-Content-Type-Options        : nosniff
Cache-Control                 : no-cache
Date                          : Tue, 27 Apr 2021 06:59:55 GMT
Server                        : Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0

Body:
{
  "id": "/subscriptions/{subscription-id}/resourceGroups/***/providers/Microsoft.Storage/storageAccounts/***/inventoryPolicies/default",
  "name": "DefaultInventoryPolicy",
  "type": "Microsoft.Storage/storageAccounts/inventoryPolicies",
  "properties": {
    "lastModifiedTime": "2021-04-27T06:59:55.7238516Z",
    "policy": {
      "enabled": false,
      "type": "Inventory",
      "rules": [
        {
          "destination": "weitest1",
          "enabled": true,
          "name": "test2",
          "definition": {
            "format": "Parquet",
            "schedule": "Weekly",
            "objectType": "Blob",
            "schemaFields": [
              "Name",
              "Last-Modified",
              "Metadata",
              "LastAccessedTime"
            ]
          }
        }
      ]
    }
  }
}
```

HimanshuChhabra replied:
Thanks for pointing out, it is a server problem. blobTypes is always required for Blob object type and is not applicable for Container type. I have made the change and it will be deployed to stage region tomorrow.

KalpeshChavan12 committed:
[Hub Generated] Public private branch 'dev-maintenance-Microsoft.Maintenance-2021-05-01' (#14148)

* Adds base for updating Microsoft.Maintenance from version preview/2020-07-01-preview to version 2021-05-01

* Updates readme

* Updates API version in new specs and examples

* Added system data properties.

* Fixed lint error.
Added documentation.

* Use Host Maintenance scope and api version in exmaple.

Co-authored-by: Kalpesh Chavan <kachavan@microsoft.com>

binyu91 committed:
Add Test Base Swagger Spec from RPSaaSMaster (#14113)

Co-authored-by: Bin Yu <biy@microsoft.com>

bar810 committed:
rename property: change category to categories (#13883)

harini-ms committed:
Edit description of Update Iot Hub Resource Api (#13080)

Co-authored-by: Harini <codebyh.365@gmail.com>

chandrikagonuguntla committed:
[Hub Generated] Review request for Microsoft.RecoveryServices to add version stable/2021-03-01 (#13889)

* Adds base for updating Microsoft.RecoveryServices from version stable/2018-07-10 to version 2021-03-01

* Updates readme

* Updates API version in new specs and examples

vicancy committed:
Update webpubsub.json (#14096)

bowgong committed:
add dataFeedId in AnomalyResult and IncidentResult (#14127)

* add dataFeedId in AnomalyResult and IncidentResult

* fix examples

alexandra142 committed:
Add new version 2021-03-31-preview1 for ACS Auth (#13965)

* Add new version 2021-03-31-preview

* Set required body

* Add all paths to new version

* Update description for property of TeamsAccessTokenRequest model

* Change version from 2021-03-31-preview to 2021-03-31-preview1 following standards https://skype.visualstudio.com/SPOOL/_wiki/wikis/SPOOL.wiki/16790/Versioning

* Rename folder

* Update method description

* remove Turn endpoint

* Add examples

* Update readme

* Omit the word "skypetoken" from swagger

* Link example

* Update description and path for a new endpoint

Co-authored-by: Alexandra Lamiova Pistrakova <apistrak@microsoft.com>

raych1 committed:
Added example gneration document links (#14151)

parvsaxena committed:
purview scanning service (#14021)

* purview scanning service

* Add custom words for spellcheck

* fix config files and add endpoint param

Co-authored-by: Parv Saxena <pasaxena@microsoft.com>
Co-authored-by: iscai-msft <iscai@microsoft.com>

stivanov-git committed:
MG RP new API versions: 2020-10-01 and 2021-04-01 (#13897)

* API 2020-10-01 with 2020-05-01 files

* 2020-10-01 API changes

* 2021-04-01 API version with 2020-10-01 files

* Changes for 2021-04-01 API

* PR fixes

* Fixed issues with model

* PR fixes

Zzegfried committed:
Latest (#13917)

jiacheng-L committed:
Pick up previous Swagger Fix in 2019-09-01 version (#14092)

* Add missing swagger fix

* Fix bad copy-paste

* update the some capitalized words in the description

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/examples/BlobContainersPutObjectLevelWorm.json:
```diff
@@ -0,0 +1,43 @@
+{
+  "parameters": {
```
Is there a region enabled for VLW.
I tried to create an account with VLW enabled in account in canary (eastus2euap), but failed with 400.

```
PUT https://management.azure.com/subscriptions/{subscription-id}/resourceGroups/***/providers/Microsoft.Storage/storageAccounts/***/blobServices/default/containers/container7176?api-version=2021-04-01 HTTP/1.1
x-ms-client-request-id: 1cbed7ec-0d47-478d-a684-68faec042d88
Accept-Language: en-US
Authorization: Bearer ***
User-Agent: FxVersion/4.6.29916.01 OSName/Windows OSVersion/Microsoft.Windows.10.0.19042. Microsoft.Azure.Management.Storage.StorageManagementClient/21.1.0.0
Content-Type: application/json; charset=utf-8
Content-Length: 99
Host: management.azure.com

{
  "properties": {
    "immutableStorageWithVersioning": {
      "enabled": true
    }
  }
}
HTTP/1.1 400 Bad Request
Cache-Control: no-cache
Pragma: no-cache
Content-Length: 131
Content-Type: application/json
Expires: -1
x-ms-request-id: aa1532ce-1409-4ad4-b1d6-4b91f4431f5f
Strict-Transport-Security: max-age=31536000; includeSubDomains
Server: Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0
x-ms-ratelimit-remaining-subscription-writes: 1194
x-ms-correlation-request-id: 67dc00fc-b0b2-4173-b54e-bbf9bcb1720e
x-ms-routing-request-id: SOUTHEASTASIA:20210428T092733Z:67dc00fc-b0b2-4173-b54e-bbf9bcb1720e
X-Content-Type-Options: nosniff
Date: Wed, 28 Apr 2021 09:27:32 GMT

{"error":{"code":"FeatureNotAvailableInRegion","message":"ImmutableStorageWithVersioning feature is not available in this region"}}
```

Neku-microsoft committed:
Adding EventData schema for Microsoft.Storage.BlobTierChanged event (#14049)

* Adding EventData schema for Microsoft.Storage.BlobTierChanged event

* Removing contentLength/Offset from Microsoft.Storage.BlobTierChanged data schema

* Removing BlobTierInfo from BlobTierChanged and adding new event Microsoft.Storage.AsyncOperationInitiated

* adding ContentLength for BlobTierChanged and AsyncOperationInitiated events

giakas committed:
[Hub Generated] Review request for VideoAnalyzer.Edge to add version preview/1.0.0 (#13682)

* New Readme Config File

* New Go Language Readme Config File

* New Azure AZ Readme Config File

* New Azure CLI Readme Config File

* New Typescript Language Readme Config File

* New Python Language Readme Config File

* New C# Language Readme Config File

* New AzureResourceSchema Readme Config File

* New Swagger Spec File

* New Swagger Example Spec File

* Copy old LVA swaggers

* Rename old swaggers to new file names

* Update LVA swaggers with new AVA swagger

* fixed some paths and capitalization

* fix prettier check issues

* fixing a Pascal casing

* Updating swaggers to latest

* fix linting issue

* removed an unused definition

jeffj6123 committed:
Service fabric Client API 8.0 (#14011)

* spell checked and linted.

* updated tag version.

* reverted change outside of SF folder.

* updated version in swagger.

* changed credential issues.

* changed 1 more that looks like a credential.

chiragg4u commented:
Please go through the contribution checklist for ARM API Review added as part of the review and mark them completed. It will help you fix most of the issues and improve the quality of swagger and help close the PR faster. This is a requirement before review can be continued.

ItaySofer committed:
Kusto pools principal assignment (#14079)

* kustoPool PrincipalAssignments

* fixed list principal assignments

* principalAssignment list example

* principalAssignment get example

* KustoPoolPrincipalAssignmentsCreateOrUpdate

* KustoPoolPrincipalAssignmentsDelete

* databases PrincipalAssignments

* minor change

* KustoPoolDatabasePrincipalAssignmentsList

* KustoDatabasePrincipalAssignmentsGet

* KustoPoolDatabasePrincipalAssignmentsCreateOrUpdate

* KustoPoolDatabasePrincipalAssignmentsDelete

* fixed example ref

* added system data

* kustoPoolCheckNameAvailability

* KustoPoolsCheckNameAvailability example

Co-authored-by: Itay Sofer <itsofer@microsoft.com>

LukeSlev committed:
[Service Fabric Managed Clusters] fix api 2021_01_01-preview (#14012)

* Update preview api to disable broken properties and to add property that is working

* Update managed cluster description

yanfa317 committed:
Add New Data Model RuleCollection to NetworkManager (#13827)

* Add New Data Model RuleCollection to NetworkManager

* fix typo

* test

* test

* Fix discriminator

* fix type

* Fix discriminator

* Fix type

* fix type

* Fix type

* Fix discriminator

* Fix discriminator

* fix type

* fix type

* Fix bug

* Fix type

* Fix format

* Fix name

* Fix name

* fix spell

* fix description

* Fix spell

* change to kind

* fix lint

* fix lint

* fix lint

* fix lint

* fix lint

* fix lint

* Fix post body

* Fix as comments

* fix bug

* fix bug

* fix bug

* fix bug

* fix bug

* fix bug

* change rule kind

* Fix as comments

* Fix as comments

* update lastest version

* Update type

* Update Skiptoken

* Fix discription

* Fix NG

* Fix NG

* add more example

msyyc committed:
[python]Update readme.python.md for postgresql (#14182)

* Update readme.python.md

* Update readme.python.md

leniatgh committed:
Remove all schema readme (#13885)

* Remove all schema readme

* process new rp

* Remove after_scripts

Juliehzl reviewed (changes requested):
@HimanshuChhabra Please add your new api version in readme. Here is a reference https://github.com/Azure/azure-rest-api-specs/pull/13550/commits/c0c5fcdc64b8e53f9ac5e154ac2e4dc28e35b7f4

dishonono committed:
[Hub Generated] Review request for Microsoft.AlertsManagement to add version stable/2021-04-01 (#13890)

* Adds base for updating Microsoft.AlertsManagement from version stable/2019-06-01 to version 2021-04-01

* Updates readme

* Updates API version in new specs and examples

* first commit

* some fixes

* try without alert management

* revert change to name of namespace

* Resolve lintdiff issues

* remove the copy of files and reference alert managment previous version

* update description of subscriptionId to solve error

* attemp to remove AlertsManagement from the package

* revert (put back AlertsManagment)

ArcturusZhang committed:
add tags for go SDK release v54.0.0 (#14161)

* add tags for go SDK release v54.0.0

* add a new tag

sandeepparab committed:
FRv2.1 GA Stable Release (#13997)

* Standardized Errors, sample list.

* removing from master from now as this is just a proposal

* FRv2.1 GA Initial commit.

Co-authored-by: Sandeep Parab <rparab@ntdev.microsoft.com>

kryalama committed:
remove extra / from host template (#14176)

dpwatrous committed:
[Hub Generated] Review request for Microsoft.Batch to add version stable/2020-09-01.12.0 (#14170)

* [Batch] Update RemoveNodes description to include max nodes/request

* Added suppressions to readme

alexandra142 committed:
Update swagger for a new version 2021-03-31-preview1 for ACS Auth (#14168)

* Unify names

* Update tag

* Update due to review meeting

* update example description

Co-authored-by: Alexandra Lamiova Pistrakova <apistrak@microsoft.com>

yang-hai-feng committed:
Add new 1.0 version for IoT Central API (#13713)

* Add new 1.0.0 version for IoT Central API

* fix swagger style

* fix wrong json file name

* remove unused examples

* revert back email format

* remove non-used examples

* update with latest examples

* update preview from 2019-10-28-preview to 2021-04-30-preview

* remove extra ---

* update version from 1.0.0 to 1.0

* rollback collections for common entities

* update _id to Id, update _name to Name from operation, add sdk generation

* fix go sdk file

* remove go sdk

* updatedevicve_templateId to deviceTemplateId

* Add query parameter based versioning, remove resource schema

* fix example docs reference

* fix typo in comments

* renaming execute to run for command operations

* add api-version to samples files, fix devices_update missining issue

* add devices_update.json for 1.0

* fix rerun_id, add missing parameters to examples for preview, fix command comments from execute to run

* fix annoymous type issues

giakas committed:
add title to readme. (#14179)

weidongxu-microsoft committed:
videoanalyzers-fix-missing-host (#14181)

blueww commented on specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/file.json:
```diff
@@ -0,0 +1,1063 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/fileServices/default/shares/{shareName}/lease": {
```
@zfchen95 
As we discussed in the mail, should lease snapshot.

wwendyc committed:
New API version for Microsoft.Resources stable/2021-04-01 (#14132)

* base for Microsoft.Resources 2021-04-01

* update readme

* update api version

* API changes for S2S

* readme

Co-authored-by: Wendy Chang <wendychang@gmail.com>

pradeepkintali committed:
Swagger Changes for Get Recoverable Servers and create server with Geo-Restore. (#14013)

* Added Recoverable Servers API and GeoRestore API.

* Fixed spell check and prettier

* Fixed readme.md file issue and changed the postgreSQL to restore.

* Fixed duplicate schema issue.

* Fixed the mismatch with namespace names.

* change the tag name from georestore to servers

* Fixed the SDK validation issues.

* Fixed objects w.r.t preview version

* Fixed operation required properties.

* Included 2020-04-10 in the SDK versions

* Change the folder path.

ArcturusZhang committed:
Add missing multiapi entry for go sdk v54.0.0 release (#14200)

* add missing multiapi entry

* fix one output-folder

sagamzu committed:
Azure security insights - add data connectors preview version (#13998)

* add new API

* ad name to enum

* updatre defenition names

* update readme

* update readme

* update readme path files

* fix errors

* add missing connectors

* execute validation

* execute pre

* update readme

* add description ans x-ms-enum

* update description

* update properties

* add description

* rename enum - fix new error that didnt display until now

* rename enum unser x-ms-enum property

JCookit committed:
[Hub Generated] Review request for Microsoft.Maps/Alias to add version preview/2.0, preview/2.0, preview/2.0, preview/2.0, preview/2.0, preview/2.0 and preview/2.0 (#13921)

* Copy Creator API 1.0 to 2.0

* draft of planB

* adding ontology properties; adding DataUpload

* updated alias apis

* put v1 API back so diffing is easier

* reapplying v2 so diffs are easier

* remove type from datasetsources

* incorporate suggestions from review board

* adding ontology field

* Update alias v2 apis.

* fix conversion descriptions, update alias

* Fix the alias file.

* Add get details example.

* fix casing on headers; mapData more consistent

* minor comment and linting fixes

* Adds base for updating Microsoft.Maps/Alias from version preview/1.0 to version 2.0

* Updates readme

* Updates API version in new specs and examples

* Add baseline from preview/1.0

* update api-version in examples

* add v2 files to readme

* fix avacado errors

* Adding string length limit to the descriptions for feature state.

Adding string length limit to the descriptions for feature state.

* Adding ontology element to the WFS resopnse examples.

Adding ontology element to the WFS resopnse examples.

* Updating length to 1000 from 1024.

Updating length to 1000 from 1024.

* Update dataset v2 descriptions and examples.

* Update the examples to have the right details.

* reran linter; added spellcheck words

* more linting: 202 description alignment

* fixed example linting for alias, data, dataset

* example linting changes for conv, fs, tileset

* change featurestateset patch to put

* autorest linting fixes

* Reverting FeatureState body to include array to match V1 body and response.

* adding missing operationIds

* added examples; more linting

* remove x-ms-long-running

* Update alias documentation for resourceId -> creatorDataItemId change

Co-authored-by: Robert Johnson <rojohn@microsoft.com>
Co-authored-by: Subbarayudu Kamma <skamma@microsoft.com>

pohtsng committed:
Update attribute name for message metadata (#14206)

* Update attribute name for message metadata

* Apply Prettier

* Prettier fix

Co-authored-by: Po Tsang <tsangpo@microsoft.com>

HimanshuChhabra committed:
Add 201, 202 response types to FileshareLease Acquire and Break functions, updated examples

HimanshuChhabra committed:
Add 201, 202 response types to FileshareLease Acquire and Break functions, updated examples

HimanshuChhabra committed:
Update pending readme files, correct spellcheck in file.json

HimanshuChhabra committed:
Update pending readme files, correct spellcheck in file.json

aleksbol committed:
Re-add a previously erroneously removed PUT API for /networkConfig/VirtualNetwork (#14018)

* Add CreateOrUpdateSwiftVirtualNetworkConnectionWithCheck, and site-level virtualNetworkSubnetId field

* Add site-level virtualNetworkSubnetId field

* Tabs vs spaces

* Removing the new property since it's technically a 'breaking change'

* Add changes needed for correct .NET SDK generation issues. Reference PR: https://github.com/Azure/azure-sdk-for-net/pull/20300

* Fix prettier issue.

* Remove breaking changes other than property order changes

* Make it pretty again

* Added new properties SiteConfig.PublicNetworkAccess and Site.virtualNetworkSubnetId to enable azure policy

* Fix prettier and spellcheck issue

* Fix spellcheck issue

* Add virtualNetworkSubnetId prperty to Site objec tmodel

Co-authored-by: Naveed Aziz <naveeda@microsoft.com>

HimanshuChhabra committed:
Merge branch 'newapr21swagger' of https://github.com/HimanshuChhabra/azure-rest-api-specs into newapr21swagger
